packages/aws-cdk-lib/core/lib/stack.ts

```diff
@@ -1104,11 +1107,17 @@ export class Stack extends Construct implements ITaggable {
     fs.writeFileSync(outPath, templateData);
 
     for (const ctx of this._missingContext) {
-      if (lookupRoleArn != null) {
```

Comment:
There is no need for this `if` because undefined properties are filtered anyway.